doc/dev/transforms.rst

```diff
@@ -0,0 +1,574 @@
+
+            Value argPlusH = rewriter.create<arith::AddFOp>(loc, arg, hValue);
+
+            // Make a copy of arguments to replace the argument with it's shifted value.
+            std::vector<Value> callArgsForward(callArgs.begin(), callArgs.end());
+            callArgsForward[idx] = argPlusH;
+            func::CallOp callOpForward =
+                rewriter.create<func::CallOp>(loc, callee, callArgsForward);
+
+            // Compute the finite difference.
+            Value difference = rewriter.create<arith::SubFOp>(loc, callOpForward.getResult(0), callOp.getResult(0));
+            Value partialDerivative = rewriter.create<arith::DivFOp>(loc, difference, hValue);
+            gradient.push_back(partialDerivative);
+        }
+
+        rewriter.create<func::ReturnOp>(loc, gradient);
+    }
+
+Alright, our function should now look something like this:
+
+.. code-block:: llvm
+
+    func.func @my_func.finitediff(%x: f64, %y: f64, %z: f64) -> (f64, f64, f64) {
+        %h = arith.constant 0.1 : f64
+
+        %fres = func.call @my_func(%x, %y, %z) : (f64, f64, f64) -> f64
+
+        %xph = arith.addf %x, %h : f64
+        %fxph = func.call @my_func(%xph, %y, %z) : (f64, f64, f64) -> f64
+        %diffx = arith.subf %fxph, %fres : f64
+        %dx = arith.divf %diffx, %h
+
+        %yph = arith.addf %y, %h : f64
+        %fyph = func.call @my_func(%x, %yph, %z) : (f64, f64, f64) -> f64
+        %diffy = arith.subf %fyph, %fres : f64
+        %dy = arith.divf %diffy, %h
+
+        %zph = arith.addf %z, %h : f64
+        %fzph = func.call @my_func(%x, %y, %zph) : (f64, f64, f64) -> f64
+        %diffz = arith.subf %fzph, %fres : f64
+        %dz = arith.divf %diffz, %h
+
+        func.return %dx, %dy, %dz : f64, f64, f64
+    }
+
+Finally, we have to amend our rewrite function to invoke the new function we created and delete the
+``GradOp`` from the IR:
+
+.. code-block:: cpp
+
+    void FiniteDiffLowering::rewrite(GradOp op, PatternRewriter &rewriter)
+    {
+        ...
+            populateFiniteDiffMethod(rewriter, op, gradFn);
+        }
+
+        rewriter.replaceOpWithNewOp<func::CallOp>(op, gradFn, op.getArgOperands());
+    }
+
+Note how we can create a new operation, take its results, and use those to replace another operation
+in one go. This turns the previous IR:
+
+.. code-block:: llvm
+
+    func.func @my_func(%x: f64, %y: f64, %z: f64) -> f64 {
+        ...
+    }
+
+    func.func @my_func.finitediff(f64, f64, f64) -> (f64, f64, f64) {
+        <contents of populateFiniteDiffMethod>
+    }
+
+    %deriv:3 = gradient.grad "fd" @my_func(%x, %y, %z) : (f64, f64, f64) -> (f64, f64, f64)
+
+into:
+
+.. code-block:: llvm
+
+    func.func @my_func(%x: f64, %y: f64, %z: f64) -> f64 {
+        ...
+    }
+
+    func.func @my_func.finitediff(f64, f64, f64) -> (f64, f64, f64) {
+        <contents of populateFiniteDiffMethod>
+    }
+
+    %deriv:3 = func.call @my_func.finitediff(%x, %y, %z) : (f64, f64, f64) -> (f64, f64, f64)
```

Comment:
The guide finishes rather abruptly 😆 Wondering if there were 1-2 sentences we could finish with? Maybe links for more information, or a call-to-action to contribute transforms?

Reply:
Hmm, I put all the informative links at the top 😅 Maybe a call to action I guess?

But if we add the section on how to add, compile, and use a new pass, the end would look different I guess. Or were you thinking to add such a section later on?

Reply:
> But if we add the section on how to add, compile, and use a new pass,

Yep was thinking of doing this later on, not now :)

Reply:
Added a small section here: https://github.com/PennyLaneAI/catalyst/pull/340/commits/6ac86fdb582ec41495afc40a4ae109250ec0b184